tests/parse_command_test.c

```diff
@@ -0,0 +1,99 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parse_command_test.c                               :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: taretiuk <taretiuk@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/10/16 16:50:45 by taretiuk          #+#    #+#             */
+/*   Updated: 2024/10/17 22:49:33 by taretiuk         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "tests_internal.h"
+#include "../sources/parsing/input_processing_internal.h"
+
+#define NUM_TEST_CASES 11
+#define MAX_ARGS 16
+
+t_strings	create_string_array()
+{
+	t_strings	str_array;
+	str_array.count = 11;
+	str_array.error = 0;
+	str_array.strs = ft_calloc_if(sizeof(t_string) * str_array.count, 1);
```

Comment:
1. no need for ft_calloc_if in tests, probably we won't be testing this
2. you can just assert(str_array.strs != 0), this would be enough for the tests. Everything will fall, but this is testing anyway, we don't have to care about gracefullness if there is no memory